Sources/MongoSwiftSync/MongoCollection.swift

```diff
@@ -38,8 +38,13 @@ public struct MongoCollection<T: Codable> {
     /// The underlying asynchronous collection.
     internal let asyncColl: MongoSwift.MongoCollection<T>
 
+    /// The client this collection was derived from. We store this to ensure it remains open for as long as this object
+    /// is in scope.
+    private let client: MongoClient
```

Comment:
i realized that my previous claim that we didn't need to do this was incorrect. for reference counting purposes we do indeed need to store references to sync clients from sync DBs/colls to ensure that their sync clients (and therefore their async clients) are kept alive automatically.